src/pages/ReimbursementAccount/CompanyStep.js

```diff
@@ -91,56 +123,74 @@ class CompanyStep extends React.Component {
+     * Clear the error associated to inputKey if found and store the inputKey new value in the state.
+     *
+     * @param {String} inputKey
+     * @param {String} value
+     */
+    clearErrorAndSetValue(inputKey, value) {
+        this.setValue({[inputKey]: value});
+        const errors = this.getErrors();
+        if (!errors[inputKey]) {
+            // No error found for this inputKey
+            return;
+        }
+
+        // Clear the existing error for this inputKey
+        const newErrors = {...errors};
+        delete newErrors[inputKey];
+        setBankAccountFormValidationErrors(newErrors);
+    }
+
     /**
      * @returns {Boolean}
      */
     validate() {
+        const errors = {};
         if (!this.state.password.trim()) {
-            showBankAccountFormValidationError(this.props.translate('common.passwordCannotBeBlank'));
-            return false;
+            errors.password = true;
         }
 
         if (!isValidAddress(this.state.addressStreet)) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.addressStreet'));
-            return false;
+            errors.addressStreet = true;
         }
 
         if (this.state.addressState === '') {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.addressState'));
-            return false;
+            errors.addressState = true;
         }
 
         if (!isValidZipCode(this.state.addressZipCode)) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.zipCode'));
-            return false;
+            errors.addressZipCode = true;
         }
 
         if (!Str.isValidURL(this.state.website)) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.website'));
-            return false;
+            errors.website = true;
         }
 
         if (!/[0-9]{9}/.test(this.state.companyTaxID)) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.taxID'));
-            return false;
+            errors.companyTaxID = true;
         }
 
         if (!isValidDate(this.state.incorporationDate)) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.incorporationDate'));
-            return false;
+            errors.incorporationDate = true;
         }
 
         if (!isValidIndustryCode(this.state.industryCode)) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.industryCode'));
-            return false;
+            errors.industryCode = true;
         }
 
         if (!this.state.hasNoConnectionToCannabis) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.restrictedBusiness'));
-            return false;
+            errors.hasNoConnectionToCannabis = true;
         }
+        this.requiredFields.forEach((inputKey) => {
+            if (!this.state[inputKey].trim()) {
```

Comment:
NAB, this works fine now, but seems potentially dangerous to assume this will always be a string e.g. if someone made a boolean a required field or changes something to an int - this would break since we can only call `.trim()` on a string.

Reply:
Makes sense, should we add a more robust function that test if it is a string? Something like:
- if it is falsey (`undefined`, `null`, whatever is falsey), return false
- If it is a `string`, use `trim`, test with `_.isEmpty`
- If it is a `array`, test with `_.isEmpty`
- return !!`this.state[inputKey]`

A function called like `isValuePresent`

Reply:
That might be useful. If we want to add it maybe it can go in the `ValidationUtils.js`? I could go either way here so will leave it up to you to decide if it's worth doing now or not.

Reply:
Added `isValuePresent` function in `ValidationUtils.js`